src/test/java/teammates/test/cases/webapi/ArchiveCourseActionTest.java

```diff
@@ -28,71 +30,67 @@ protected String getRequestMethod() {
     protected void testExecute() throws Exception {
         InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
         String instructorId = instructor1OfCourse1.googleId;
-        String courseId = instructor1OfCourse1.courseId;
 
         loginAsInstructor(instructorId);
 
         ______TS("Not enough parameters");
         verifyHttpParameterFailure();
-        verifyHttpParameterFailure(Const.ParamsNames.COURSE_ID, courseId);
-        verifyHttpParameterFailure(Const.ParamsNames.COURSE_ARCHIVE_STATUS, "true");
 
         ______TS("Typical case: archive a course");
 
         String[] submissionParams = new String[] {
                 Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
-                Const.ParamsNames.COURSE_ARCHIVE_STATUS, "true",
         };
 
-        ArchiveCourseAction archiveAction = getAction(submissionParams);
+        CourseArchiveRequest courseArchiveRequest = new CourseArchiveRequest();
+        courseArchiveRequest.setArchiveStatus(true);
+
+        ArchiveCourseAction archiveAction = getAction(courseArchiveRequest, submissionParams);
         JsonResult result = getJsonResult(archiveAction);
 
+        InstructorAttributes theInstructor = InstructorsLogic.inst().getInstructorForGoogleId(
+                instructor1OfCourse1.getCourseId(), instructor1OfCourse1.getGoogleId());
+
         assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+        assertTrue(theInstructor.isArchived);
 
         ______TS("Rare case: archive an already archived course");
 
-        submissionParams = new String[] {
-                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
-                Const.ParamsNames.COURSE_ARCHIVE_STATUS, "true",
-        };
+        courseArchiveRequest.setArchiveStatus(true);
 
-        archiveAction = getAction(submissionParams);
+        archiveAction = getAction(courseArchiveRequest, submissionParams);
         result = getJsonResult(archiveAction);
 
         assertEquals(HttpStatus.SC_OK, result.getStatusCode());
 
         ______TS("Typical case: unarchive a course, redirect to Courses page");
```

Comment:
`redirect to Courses page`???

Reply:
Artifact of copying over the previous action test  🤣